### ISSUE TITLE
Fragment short syntax <></> causes the parser to crash

### ISSUE THREAD

**Issue by KornelSzaltanowicz** [laget-se/react-gettext-parser#48]
https://github.com/laget-se/react-gettext-parser/blob/9e41e9e6235b11de648153c0da8499093566c0fd/src/node-helpers.js#L9

Somehow I have node as undefined and ofc app crashes. 
Maybe can we add some condition to check whether node is defined or so? 

simply ```if(!node) return false;``` Do the trick for me :)

**Comment by alexanderwallin** [comment]
Could you post the source code that causes this to happen?

**Comment by KornelSzaltanowicz** [comment]
This is my simple script to perform translations..
---

```
// 18n2pot.js
const fs = require("fs");
const path = require("path");
const { extractMessagesFromGlob, toPot } = require("react-gettext-parser");

//default options
let options = {
    globPaths: [path.resolve("./", "src/**/*.js")],
    resultFilePath: path.resolve("./", "translations.pot"),
    extractOptions: {
        componentPropsMap: {
            T: {
                message: "msgid",
                messagePlural: "msgid_plural",
                context: "msgctxt",
                comment: "comment"
            }
        },
        funcArgumentsMap: {
            t: ["msgid"]
        }
    }
};

const getRawMessages = () =>
    new Promise(resolve => {
        const rawMessages = extractMessagesFromGlob([...options.globPaths], options.extractOptions);
        resolve(rawMessages);
    });

const mapToPot = rawMessages =>
    new Promise(resolve => {
        const poContents = toPot(rawMessages);
        resolve(poContents);
    });

const writePotFile = messages =>
    new Promise(resolve => {
        fs.writeFile(options.resultFilePath, messages, () => {
            resolve(true);
        });
    });

const translate = () =>
    new Promise((resolve, reject) => {
        getRawMessages()
            .then(data => mapToPot(data))
            .then(data => writePotFile(data))
            .then(resolve)
            .catch(reject);
    });

const init = (customOptions = defaultOptions) => {
    for (let key in customOptions) {
        if (options.hasOwnProperty(key)) {
            if (key === "globPaths") {
                options[key] = customOptions[key].map(item => path.resolve("./", item));
            }
            if (key === "resultFilePath") {
                options[key] = path.resolve("./", customOptions[key]);
            }
            options[key] = customOptions[key];
        } else {
            console.error("Invalid property for option object: " + key);
            console.info("For more info please see documentation");
        }
    }
};

module.exports = {
    init,
    translate
};
```
------

```
// myLib.js

...
const productPath = path.join(rootPath, `./products/${PRODUCT_NAME}/src/**/*.js`);
    const corePath = path.join(rootPath, `./core/src/**/*.js`);
    var resultFilePath = path.resolve(moveToDir + "/translations.pot");

    const config = {
      globPaths: [productPath, corePath],
      resultFilePath
    };
    toPotTranslator.init(config);
    toPotTranslator.translate().then(() => {
      process.exit(0);
    });
...
```

As you can see there is nothing special around this... 
Do we need something more?

**Comment by KornelSzaltanowicz** [comment]
![screen shot 2019-01-16 at 13 46 28](https://user-images.githubusercontent.com/7700156/51249921-32866200-1995-11e9-9a07-f340af2be549.png)

**Comment by sHooKDT** [comment]
Uhm, thats interesting.
I think @alexanderwallin has meant the code we extract messages from (your app sources, not extracting scripts). It would be very nice if you can trace it and give us an example.
It's very strange that we somehow get undefined node when visiting ast.
I think it can only be cause be this
https://github.com/laget-se/react-gettext-parser/blob/9e41e9e6235b11de648153c0da8499093566c0fd/src/parse.js#L223
or
https://github.com/laget-se/react-gettext-parser/blob/9e41e9e6235b11de648153c0da8499093566c0fd/src/parse.js#L261
lines.

**Comment by alexanderwallin** [comment]
As @sHooKDT said, it would be great to see the source code. The info on the location of this error is in that `node` object you are printing out. Check the `loc` property to trace where this is and paste that code on here!

**Comment by KornelSzaltanowicz** [comment]
Ok, guys I found what was the problem: 

Not working code: 

```
<I18nContext.Consumer>
    {value => (
      <>
        <HeaderMobile onLogin={onLogin} onLogout={onLogout} user={user} {...rest} {...value} />
      ...other components
      </>
    )}
  </I18nContext.Consumer>
```
---

working code: 

```
<I18nContext.Consumer>
    {value => (
      <React.Fragment>
        <HeaderMobile onLogin={onLogin} onLogout={onLogout} user={user} {...rest} {...value} />
       ...other components
      </React.Fragment>
    )}
  </I18nContext.Consumer>
```

So we can't use <></> as fragment short syntax.

**Comment by alexanderwallin** [comment]
Oh, I haven't seen that shorthand before. Since it's part of the official React API we should support it, if possible. Let's see whether it can be parsed with babylon!

**Comment by alexanderwallin** [comment]
The issue was with the fragment's child nodes:

https://github.com/laget-se/react-gettext-parser/blob/09e14ac7c7c913895b177e4954dd14e7182d9428/src/parse.js#L223-L232

I've published 1.8.2 with a fix to npm, so go check if it works for you!

**Comment by naruaway** [comment]
@alexanderwallin Hi! This implementation still has a problem when dealing with direct text node.
I created a pull request to explain and solve the problem with a test case.
See https://github.com/laget-se/react-gettext-parser/pull/56
